src/core/shape/vertex.js

```diff
@@ -514,13 +518,33 @@ p5.prototype.bezierVertex = function(...args) {
  * @alt
  * Upside-down u-shape line, mid canvas with the same shape in positive z-axis.
  */
-p5.prototype.curveVertex = function(...args) {
+p5.prototype.curveVertex = function (...args) {
   p5._validateParameters('curveVertex', args);
   if (this._renderer.isP3D) {
     this._renderer.curveVertex(...args);
   } else {
-    isCurve = true;
-    this.vertex(args[0], args[1]);
+    let allVertObj = [];
+
+    shape.contourss.forEach(c => {
+      c.segmentss.forEach(s => {
+        s.verticess.forEach(v => {
+          allVertObj.push(v);
+        });
+      });
+    });
```

Comment:
I think this is just a [`flatMap`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap)?